daml-lf/encoder/src/main/scala/com/digitalasset/daml/lf/testing/archive/DamlLfEncoder.scala

```diff
@@ -24,46 +26,118 @@ private[digitalasset] object DamlLfEncoder extends App {
 
   private val pkgId = Ref.PackageId.assertFromString("-self-")
 
-  try {
-    val (inputFile, outputFile, languageVersion): (String, String, LanguageVersion) =
-      args.toSeq match {
-        case Seq(input, output) =>
-          (input, output, LanguageVersion.default)
-        case Seq(input, output, version) =>
-          version.split("""\.""").toSeq match {
-            case Seq("1", minor)
-                if LanguageMajorVersion.V1.supportsMinorVersion(minor) || minor == "dev" =>
-              (input, output, LanguageVersion(LanguageMajorVersion.V1, minor))
-            case _ =>
-              error(s"version '$version' not supported")
-          }
-        case _ =>
-          error("usage: damlf-as input output [version]")
-      }
-
-    implicit val parserParameters: ParserParameters[this.type] =
-      ParserParameters(
-        defaultPackageId = pkgId,
-        languageVersion = languageVersion
-      )
+  private def main() =
+    try {
+      val appArgs = parseArgs()
+
+      implicit val parserParameters: ParserParameters[this.type] =
+        ParserParameters(
+          defaultPackageId = pkgId,
+          languageVersion = appArgs.languageVersion
+        )
+
+      makeDar(readSources(appArgs.inputFiles), Paths.get(appArgs.outputFile).toFile)
+
+    } catch {
+      case e: EncodeError =>
+        error(s"Encoding error: ${e.message}")
+      case NonFatal(e) =>
+        error(s"error: ${e.getMessage}")
+    }
+
+  private def readSources(files: Seq[String]) = {
+    val builder = StringBuilder.newBuilder
+    files.foreach { file =>
```

Comment:
`files.flatMap { file => (??? : String) }(breakOut): String` should typecheck. If you're not a fan of that, go with `map` and `mkString` over this.

Reply:
fixed